authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -118,27 +127,281 @@ public List<String> policyResourceDefinesRule() {
+      AuthorizationMetadataObject newAuthzMetaObject) {
+    PathBasedMetadataObject newPathBasedMetadataObject =
+        (PathBasedMetadataObject) newAuthzMetaObject;
+    List<RangerPolicy> oldPolicies = wildcardSearchPolies(oldAuthzMetaObject);
+    List<RangerPolicy> existNewPolicies = wildcardSearchPolies(newAuthzMetaObject);
+    if (oldPolicies.isEmpty()) {
+      LOG.warn("Cannot find the Ranger policy for the metadata object({})!", oldAuthzMetaObject);
+      return;
+    }
+    if (!existNewPolicies.isEmpty()) {
+      LOG.warn("The Ranger policy for the metadata object({}) already exists!", newAuthzMetaObject);
+    }
+    oldPolicies.stream()
+        .forEach(
+            policy -> {
+              try {
+                // Update the policy name is following Gravitino's spec
+                policy.setName(getAuthorizationPath(newPathBasedMetadataObject));
+                // Update the policy resource name to new name
+                policy
+                    .getResources()
+                    .put(
+                        rangerHelper.policyResourceDefines.get(0),
+                        new RangerPolicy.RangerPolicyResource(
+                            getAuthorizationPath(newPathBasedMetadataObject)));
+
+                boolean alreadyExist =
+                    existNewPolicies.stream()
+                        .anyMatch(
+                            existNewPolicy ->
+                                existNewPolicy.getName().equals(policy.getName())
+                                    || existNewPolicy.getResources().equals(policy.getResources()));
+                if (alreadyExist) {
+                  LOG.warn(
+                      "The Ranger policy for the metadata object({}) already exists!",
+                      newAuthzMetaObject);
+                  return;
+                }
+
+                // Update the policy
+                rangerClient.updatePolicy(policy.getId(), policy);
+              } catch (RangerServiceException e) {
+                LOG.error("Failed to rename the policy {}!", policy);
+                throw new RuntimeException(e);
+              }
+            });
+  }
+
+  /**
+   * If remove the SCHEMA, need to remove these the relevant policies, `{schema}`, `{schema}.*`,
+   * `{schema}.*.*` <br>
+   * If remove the TABLE, need to remove these the relevant policies, `{schema}.*`, `{schema}.*.*`
+   * <br>
+   * If remove the COLUMN, Only need to remove `{schema}.*.*` <br>
+   */
+  @Override
+  protected void doRemoveMetadataObject(AuthorizationMetadataObject authzMetadataObject) {
+    if (authzMetadataObject.type().equals(SCHEMA)) {
+      doRemoveSchemaMetadataObject(authzMetadataObject);
+    } else if (authzMetadataObject.type().equals(TABLE)) {
+      doRemoveTableMetadataObject(authzMetadataObject);
+    } else if (authzMetadataObject.type().equals(COLUMN)
+        || authzMetadataObject.type().equals(PATH)) {
+      removePolicyByMetadataObject(authzMetadataObject);
```

Comment:
Why this is remove instead of doRemove?

Reply:
Because we can directly call `removePolicyByMetadataObject()` function to process `COLUMN` and `PATH` case.

Reply:
The style is a little weird although it works.

Reply:
OK, unified function name.